sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -270,6 +270,8 @@ private[sql] object PartitioningUtils {
   private val upCastingOrder: Seq[DataType] =
     Seq(NullType, IntegerType, LongType, FloatType, DoubleType, StringType)
 
+  val validPartitionColumnTypes: Set[DataType] = upCastingOrder.toSet
```

Comment:
BTW, I think you have to make `validPartitionColumnTypes` a method rather than a `Set[DataType]` since `DecimalType` is not a singleton.